backend/species.py

```diff
@@ -0,0 +1,65 @@
+from app import app, species_db, users_db
+from flask import request
+
+# login part
+logged_users = set()
+
+@app.route("/add-species", methods=["POST"])
+def add_species():
+    # Necessary forms
+    name = request.form["name"]
+    water_requirements = request.form["water_requirements"]
+    required_size:float = request.form["required_size"]
+
+    # Species object creation and insertion
+    obj = {
+        "name": name,
+        "water_requirements": water_requirements,
+        "required_size": required_size,
+        "list_of_natural_aggressors": []
+    }
+    species_db.insert_one(obj)
+
+    return "Success", 200
+
+### DOESN'T SEEM TO WORK PROPERLY
+@app.route("/add-species-aggressor", methods=["POST"])
+def add_species_aggressor():
+    # Necessary forms
+    subject_name = request.form["subject_name"]
+    aggressor_name = request.form["aggressor_name"]
+
+    # Query validation
+    species_names = get_species_names()
+    if(species_names.count(subject_name) != 1):
+        return "Invalid aggression subject name", 419
+    if(species_names.count(aggressor_name) != 1):
+        return "Invalid aggressor name", 419
+
+    # Aggressor insertion
+    users_db.find_one_and_update({
```

Comment:
Aggression should be both sided, right?

Reply:
Should it? I interpreted the variable's name as a list of hostile predators, rather than both predators and prey.